apps/core/components/passengerForm/PassengerForm.js

```diff
@@ -238,32 +132,38 @@ class PassengerForm extends React.Component<Props, State> {
     this.setState({ passportId });
   };
 
+  handleBirthDateSubmit = (dateOfBirth: Date) => {
+    this.setState({ dateOfBirth });
+  };
+
   handleSavePress = () => {
-    const dateOfBirth = this.handleDateValidation();
-    if (dateOfBirth) {
-      const {
-        nationality,
-        passportId,
-        lastName,
-        name,
-        gender,
-        bags,
-        id,
-      } = this.state;
+    this.setState({ getDateRequest: true });
+  };
 
-      const newPassenger = {
-        nationality,
-        passportId,
-        dateOfBirth,
-        gender,
-        name,
-        lastName,
-        bags,
-        id: id || generateId(),
-        type: 'adult', // @TODO get type base on dateOfBirth
-      };
-      this.props.onRequestSave(newPassenger);
-    }
+  handleSave = () => {
+    const {
+      nationality,
+      passportId,
+      lastName,
+      name,
+      gender,
+      bags,
+      id,
+      dateOfBirth,
+    } = this.state;
+    const newPassenger = {
+      nationality,
+      passportId,
+      dateOfBirth,
+      gender,
+      name,
+      lastName,
+      bags,
+      id: id || generateId(),
```

Comment:
Can you put `??` instead of `||`? if the `id` is 0 it will still be treated as null this way...

Reply:
I didnt do this change, so probably shouldnt modify